jmespath/api.go

```diff
@@ -0,0 +1,24 @@
+package jmespath
+
+// NewLexer creates a new JMESPath lexer.
+func NewLexer() *Lexer {
+	lexer := newLexer()
+	return lexer
+}
+
+// NewParser creates a new JMESPath parser.
+func NewParser() *Parser {
+	parser := newParser()
+	return parser
+}
+
+// Search evaluates a JMESPath expression against input data and returns the result.
+func Search(expression string, data interface{}) (interface{}, error) {
```

Comment:
It would be nice to have a form of Search which took an IO reader. This would be helpful so users would not need to decode the JSON themselves prior to using the Search.